permissions/TemplatesDir.permission

```diff
@@ -0,0 +1,11 @@
+# -*- mode: sh -*-
+
+# x-sailjail-translation-catalog = sailjail-permissions
+# x-sailjail-translation-key-description = permission-la-templatesdir
+# x-sailjail-description = Templates directory
+# x-sailjail-translation-key-long-description = permission-la-templatesdir_description
+# x-sailjail-long-description = Access Templates directory
+
+mkdir       ${HOME}/Templates
+noblacklist ${HOME}/Templates
+whitelist   ${HOME}/Templates
```

Comment:
I'd not like to promote "Templates" directory (same applies to Desktop which you don't have here). Public is somewhat reasonable.

Reply:
Not that harmful either but mkdir + blacklist in Base makes sense.

Reply:
It is a part of standard Linux desktop. There is a reason to ignore `Desktop` as we handle desktop items differently. There is no real reason to exclude `Templates` especially since if some future partner/client expects it to be there as per to normal XDG user directories, e.g., [Arch ref](https://wiki.archlinux.org/index.php/XDG_user_directories).

Reply:
Correct but intention is to simplify basic use case. If need surfaces later, simple to add.

Reply:
Well this is included in the main `UserDirs` permission. I like to think ahead, not to my feet.

Reply:
My reasoning here is that if you get the "well known dirs" permission - which effectively means the XDG user directories, you should get the access to all of them that are applicable in our OS.

Reply:
And we do have `xdg-user-dir` tool, which [supports these](https://manpages.debian.org/stretch/xdg-user-dirs/xdg-user-dir.1.en.html).

Reply:
Maybe in our case the DESKTOP should actually point in the config to the actual location of desktop files: `$HOME/.local/share/applications` ? So we could include `Desktop` as such and fully support `xdg-user-dir` tool with these permissions as well?

Reply:
That same reasoning, that we don't use Desktop directory so it's not needed here, could be applied to Templates as well. We don't use it for anything really and allowing it could make someone think that it's used somewhere. Having Public directory as permission is easier to reason about, it's kind of general place of sharing data between apps when nothing else applies.

I don't think xdg desktop should point to $HOME/.local/share/applications because that's a different thing. It's not desktop directory, it's place to put desktop files that are shown in menu. Although, IIRC, we don't use that either, only the system wide directory.

Reply:
It is not a good argument that we do not use it but it exists in standard dirs. If you think a Linux application developer in general and maybe porting of an app to SFOS, then it may be that those dirs are used/required.

`$HOME/.local/share/applications` is not excluded in use but still it works fine and has always worked, I've seen some Harbour apps using it. And I've used it as well to add custom items to our "desktop" view.

Reply:
It is just bad to give an assumption that with these permissions you get access to all normal dirs but except you don't. Lets not talk about "well known directories" then. Lets talk about Sailfish OS accepted user home resources?

Reply:
We are here defining from our point of "well known directories". The UserDirs permission do not even need to be bound to .config/user-dirs.dirs by any means. It's matter of the Permission itself and how do we define it.

Reply:
@rainemak Um no, simply NO! We are not defining something that is quite commonly accepted to be our own version of something else. I'm not going to be a part of it.

Reply:
Lets put it this way, since, for example, `xdg-user-dir` tool is supplied by us it is kind of ridiculous that if application wants `UserDirs` permissions and then can just use a portion of the dirs that are available this tool. It confuses the developer, especially if the dirs are used to retrieve with the tool. One could expect that what that tool provides are usable with the same permissions, right? I see no counter argument to this than "we don't want to" and for me it is not enough.

We should also modify the `Desktop` entry as there is a real argument on not using that. Maybe the .config/user-dirs.dirs should be modified.

Reply:
On _normal_ Linux desktop one can put any document there to be used as a template for a new document to be created. Easy to try, just put a document there and use "Create document" anywhere on your system.

There was some talk earlier about an improved office suite made by a partner? We could even utilize/support this ourselves in new document creation. To be following the same line as _normal_ Linux. But if we wish to deviate again...

Reply:
In the unlikely event that there would be seen big enough benefit, the question can be always revisited. At the moment we don't even have much support for creating new files (well, happens if you take a picture or download a file, but that's implicit).

For app usage, as also mentioned I can use the template from the file manager, but for doing new file inside the app at least LibreOffice doesn't pick up files from templates at all and uses its own template system. I think that's all for the better and would suggest similar thing for other apps. IMO the whole template directory thing is pretty useless.

Reply:
Lets deviate then. Changed the PR.